File: hw_4_customview/src/main/java/com/gmail/elnora/fet/hw_4_customview/SettingActivity.java
package com.gmail.elnora.fet.hw_4_customview;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.ImageButton;
import android.widget.Switch;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class SettingActivity extends AppCompatActivity {
    private static final String PREF_SAVE_KEY = "TOAST_SNACK";
    private static final String PREF_NAME = "PrefSettingActivity";
    private Switch switchShowMessage;
    private ImageButton buttonBack;
    private boolean onOff = false;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        onOff = loadSetting();
        switchShowMessage = findViewById(R.id.switchShowMessage);
        switchShowMessage.setChecked(onOff);

        setOnCheckedChangeListener();
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    private void setOnCheckedChangeListener() {
        switchShowMessage.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
                onOff = isChecked;
                saveSetting(onOff);
            }
        });
    }

    private void saveSetting(Boolean isChecked) {
        SharedPreferences shared = getSharedPreferences(PREF_NAME, MODE_PRIVATE);
        SharedPreferences.Editor edit = shared.edit();
        edit.putBoolean(PREF_SAVE_KEY, isChecked);
        edit.apply();
    }

    private boolean loadSetting() {
        SharedPreferences shared = getSharedPreferences(PREF_NAME, MODE_PRIVATE);
        return shared.getBoolean(PREF_SAVE_KEY, false);
    }

}

File: hw_8_weather_app/src/main/java/com/gmail/elnora/fet/hw_8_weather_app/SettingActivity.kt
package com.gmail.elnora.fet.hw_8_weather_app

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_setting.toolbar
import kotlinx.android.synthetic.main.activity_setting.viewSwitchUnit

class SettingActivity : AppCompatActivity() {
    private var unit = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)
        setSupportActionBar(toolbar)
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
        unit = loadSetting()
        viewSwitchUnit.isChecked = unit
        setOnCheckedChangeListener()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun setOnCheckedChangeListener() {
        viewSwitchUnit!!.setOnCheckedChangeListener { _, isChecked ->
            saveSetting(isChecked)
        }
    }

    private fun saveSetting(isChecked: Boolean) {
        val shared = getSharedPreferences(PREF_NAME, MODE_PRIVATE)
        val edit = shared.edit()
        edit.putBoolean(PREF_SAVE_KEY, isChecked)
        edit.apply()
    }

    private fun loadSetting(): Boolean {
        val shared = getSharedPreferences(PREF_NAME, MODE_PRIVATE)
        return shared.getBoolean(PREF_SAVE_KEY, false)
    }

    companion object {
        const val PREF_SAVE_KEY = "METRIC_IMPERIAL"
        const val PREF_NAME = "PrefSettingActivity"
    }
}